imap_processing/swe/tests/test_swe_cem_raw.py

```diff
@@ -0,0 +1,44 @@
+import pandas as pd
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.swe.l0 import decom_swe
+
+
+@pytest.fixture(scope="session")
+def decom_test_data():
+    """Read test data from file"""
+    packet_1_file = (
+        f"{imap_module_directory}/swe/tests/20230927100425_SWE_CEM_RAW_packet.bin"
+    )
+    packet_2_file = (
+        f"{imap_module_directory}/swe/tests/20230927100426_SWE_CEM_RAW_packet.bin"
+    )
+    first_data = decom_swe.decom_packets(packet_1_file)
+    second_data = decom_swe.decom_packets(packet_2_file)
+
+    return first_data + second_data
+
+
+def test_swe_raw_cem_data(decom_test_data):
+    """This test and validate raw data of SWE raw CEM data."""
+    # read validation data
+    test_data_path = f"{imap_module_directory}/swe/tests"
+    raw_validation_data = pd.read_csv(
+        f"{test_data_path}/data_raw.SWE_CEM_RAW_20230927_094839.csv",
+        index_col="SHCOARSE",
+    )
+    data = decom_test_data[0]
+    validation_data = raw_validation_data.loc[data.data["SHCOARSE"].raw_value]
+
+    # should have two packets
+    assert len(decom_test_data) == 2
+
+    # compare raw values of housekeeping data
+    for key, value in data.data.items():
+        if key == "SHCOARSE":
+            # compare SHCOARSE value
+            assert value.raw_value == validation_data.name
+            continue
+        # check if the data is the same
+        assert value.raw_value == validation_data[key]
```

Comment:
I find the continue confusing in a test case. I think you might be able to do something like this to keep it a one-liner...

```suggestion
    for key, value in data.data.items():
        # check if the data is the same, for SHCOARSE we need the name of the column
        assert value.raw_value == (validation_data[key] if key != "SHCOARSE" else validation_data.name)
```

Alternatively, I think the reason this is happening is because pandas removed it from the main columns to make the index, so maybe parsing the incoming data with `index_col=False` and then doing `raw_validation_data[raw_validation_data["SHCOARSE"] == data.data["SHCOARSE"].raw_value]` for your lookup above.

Reply:
yes. That was the reason for doing that if statement. I like that one line statement. I will add this comment about why we had to do it this way.